test/e2e/tls_test.go

```
@@ -0,0 +1,150 @@
+		data["antrea-agent.conf"] = antreaAgentConf
+	}, true, true); err != nil {
+		t.Fatalf("Failed to configure Cipher Suites and TLSMinVersion: %v", err)
+	}
+}
+
+func (data *TestData) checkTLS(t *testing.T, podName string, containerName string, apiserver int, apiserverStr string) {
+	// 1. TLSMaxVersion unset, then a TLS1.3 Cipher Suite should be used.
+	stdouts := data.openssl(t, podName, containerName, false, apiserver)
+	for _, stdout := range stdouts {
+		oneTLS13CS := false
+		for _, cs := range opensslTLS13CipherSuites {
+			if strings.Contains(stdout, fmt.Sprintf("New, TLSv1.3, Cipher is %s", cs)) {
+				oneTLS13CS = true
+				break
+			}
+		}
+		assert.Equal(t, true, oneTLS13CS,
+			"Cipher Suite used by %s Apiserver should be a TLS1.3 one, output: %s", apiserverStr, stdout)
+	}
+
+	// 2. Set TLSMaxVersion to TLS1.2, then TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 should be used
+	stdouts = data.openssl(t, podName, containerName, true, apiserver)
+	for _, stdout := range stdouts {
+		assert.Equal(t, true, strings.Contains(stdout, fmt.Sprintf("New, TLSv1.2, Cipher is %s", cipherSuiteStr)),
+			"Cipher Suite used by %s Apiserver should be the TLS1.2 one '%s', output: %s", apiserverStr, cipherSuiteStr, stdout)
+	}
+}
+
+func (data *TestData) openssl(t *testing.T, pod string, container string, tls12 bool, port int) []string {
+	var stdouts []string
+	tests := []struct {
+		enabled bool
+		ip      string
+		option  string
+	}{
+		{
+			clusterInfo.podV4NetworkCIDR != "",
+			"127.0.0.1",
+			"-4",
+		},
+		{
+			clusterInfo.podV6NetworkCIDR != "",
+			"::",
+			"-6",
+		},
+	}
+	for _, tc := range tests {
+		if !tc.enabled {
+			continue
+		}
+		cmd := []string{"timeout", "1", "openssl", "s_client", "-connect", net.JoinHostPort(tc.ip, fmt.Sprint(port)), tc.option}
+		if tls12 {
+			cmd = append(cmd, "-tls1_2")
+		}
+		stdout, _, _ := data.runCommandFromPod(antreaNamespace, pod, container, cmd)
```

ignoring errors on purpose?

---

Yes, since we just need to the check Cipher Suite used.

---

still it may be better to fail earlier if the command cannot be run for any reason (e.g. the Pod has crashed)

---

Got it. Better to be prepared for everthing. Updated.